examples/lib/stories/collision_detection/quadtree_example.dart

```diff
@@ -0,0 +1,434 @@
+  ''';
+
+  static const mapSize = 300;
+  static const bricksCount = 8000;
+
+  @override
+  Future<void> onLoad() async {
+    super.onLoad();
+
+    const mapWidth = mapSize * tileSize;
+    const mapHeight = mapSize * tileSize;
+    initCollisionDetection(
+      mapDimensions: const Rect.fromLTWH(0, 0, mapWidth, mapHeight),
+      minimumDistance: 10,
+    );
+
+    final random = Random();
+    final spriteBrick = await Sprite.load(
+      'retro_tiles.png',
+      srcPosition: Vector2.all(0),
+      srcSize: Vector2.all(tileSize),
+    );
+
+    final spriteWater = await Sprite.load(
+      'retro_tiles.png',
+      srcPosition: Vector2(0, tileSize),
+      srcSize: Vector2.all(tileSize),
+    );
+    for (var i = 0; i < bricksCount; i++) {
+      final x = random.nextInt(mapSize);
+      final y = random.nextInt(mapSize);
+      final brick = Brick(
+        position: Vector2(x.toDouble() * tileSize, y.toDouble() * tileSize),
+        size: Vector2.all(tileSize),
+        priority: 0,
+        sprite: spriteBrick,
+      );
+      add(brick);
+      staticLayer.components.add(brick);
+    }
+
+    staticLayer.reRender();
+    camera.viewport = FixedResolutionViewport(Vector2(500, 250));
+    final playerPoint = Vector2.all(mapSize * tileSize / 2);
+
+    final player =
+        Player(position: playerPoint, size: Vector2.all(tileSize), priority: 2);
+    add(player);
+    this.player = player;
+    camera.followComponent(player);
+
+    final brick = Brick(
+      position: playerPoint.translate(0, -tileSize * 2),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteBrick,
+    );
+    add(brick);
+    staticLayer.components.add(brick);
+
+    final water1 = Water(
+      position: playerPoint.translate(0, tileSize * 2),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water1);
+
+    final water2 = Water(
+      position: playerPoint.translate(tileSize * 2, 0),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water2);
+
+    final water3 = Water(
+      position: playerPoint.translate(-tileSize * 2, 0),
+      size: Vector2.all(tileSize),
+      priority: 0,
+      sprite: spriteWater,
+    );
+    add(water3);
+
+    add(
+      QuadTreeDebugComponent(
+        collisionDetection as QuadTreeCollisionDetection,
+      ),
+    );
+    add(LayerComponent(staticLayer));
+    add(FpsTextComponent());
+    camera.zoom = 1;
+  }
+
+  final elapsedMicroseconds = <double>[];
+
+  late Player player;
+  final staticLayer = StaticLayer();
+  bool firstRender = true;
+  static const stepSize = 1.0;
+
+  @override
+  KeyEventResult onKeyEvent(
+    RawKeyEvent event,
+    Set<LogicalKeyboardKey> keysPressed,
+  ) {
+    Vector2? displacement;
+    var fireBullet = false;
+    for (final key in keysPressed) {
+      if (key == LogicalKeyboardKey.keyW && player.canMoveTop) {
+        displacement = Vector2(0, -stepSize);
```

Comment:
We shouldn't encourage creating new objects within these kind of methods, use a temporary private vector and use `setFrom`/`setValues` on it instead.